### PR TITLE
CSRF support for Chatbot FE

### PR DESCRIPTION
**Pull request by ghost**
## Description
Ticket: department-of-veterans-affairs/covid19-chatbot#178
Adds a GET request in order to guarentee a CSRF token is present for creating valid chatbot tokens.

## Testing done
None

## Screenshots
N/A

## Acceptance criteria
- [x] There should be an X-CSRF token in the headers of the POST request to create a token for a corona virus chatbot session.

### CODE REVIEW

File: src/applications/coronavirus-chatbot/index.js [hunk 1]
```diff
@@ -129,14 +132,19 @@ export const requestChatBot = loc => {
...
+        .then(({ token }) => initBotConversation(token))
+        .catch(error => {
+          Sentry.captureException(error);
+          recordEvent({
+            event: `${GA_PREFIX}-connection-failure`,
+            'error-key': 'XX_failed_to_init_bot_convo',
+          });
+        }),
+    )
+    .catch(error => Sentry.captureException(error));
```
[the remark this answers is not in the document]
Comment (shaunhubbard-va) [reply src/applications/coronavirus-chatbot/index.js]: This Sentry call is not a value add to the ChatBot team. So removed.

File: src/applications/coronavirus-chatbot/index.js [hunk 2]
```diff
@@ -129,14 +132,19 @@ export const requestChatBot = loc => {
...
-        event: `${GA_PREFIX}-connection-failure`,
-        'error-key': 'XX_failed_to_init_bot_convo',
-      });
-    });
+  return ensureCSRFTokenIsSet()
+    .then(() =>
+      apiRequest(path, { method: 'POST' })
+        .then(({ token }) => initBotConversation(token))
+        .catch(error => {
+          Sentry.captureException(error);
```
[the remark this answers is not in the document]
Comment (shaunhubbard-va) [reply src/applications/coronavirus-chatbot/index.js]: This Sentry call is not a value add to the ChatBot team. So removed.

**Comment by shaunhubbard-va** [comment]
because of https://dsva.slack.com/archives/C30LCU8S3/p1588347613218700

**Comment by jholton** [comment]
> because of https://dsva.slack.com/archives/C30LCU8S3/p1588347613218700

🤔 What does that ☝️ have to do with this PR?  You still need to ensure a CSRF token is present.

**Comment by shaunhubbard-va** [comment]
@jholton merge conflicts in particular, reopening now. The aforementioned issue is out of the way.

**Pull request closed without merging**